code/clue.js

```diff
@@ -1,59 +1,495 @@
+  resident: "Patio",
+  image: "img/axe-159659_640.png",
+};
+
+const bat = {
+  id: "bat",
+  name: "Bat",
+  weight: 1.5,
+  resident: "Billiard room",
+  image: "img/base-25755_640.png",
+};
+
+const trophy = {
+  id: "trophy",
+  name: "Trophy",
+  weight: 10,
+  resident: "Lounge",
+  image: "img/cup-1757500_640.png",
+};
+
+const pistol = {
+  id: "pistol",
+  name: "Pistol",
+  weight: 2,
+  resident: "Theater",
+  image: "img/gun-5517424_640.png",
+};
+
+// Groups for all the suspects, weapons and rooms
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
-]
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
+  mrsWhite,
+];
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
+
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-const weapons = []
+// Empty object to store the randomly picked killer, weapon and room.
+let mystery = {
+  killer: "",
+  weapon: "",
+  room: "",
+};
 
-const rooms = []
+// Randomly select an item from an array
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+// Fill an array with similar objects
+const fillArray = (item, amountOfTimes) => {
+  let result = [];
+  for (let i = 0; i < amountOfTimes; i++) {
+    result.push(item);
+  }
+  return result;
+};
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// Create new weapon array if suspect has a favourite weapon
+const favouriteWeaponArray = (weapons) => {
+  if (mystery.killer.favouriteWeapon) {
+    const newWeaponArray = [
+      ...weapons,
+      ...fillArray(mystery.killer.favouriteWeapon, 5),
+    ];
+    return newWeaponArray;
+  } else {
+    return weapons;
+  }
+};
 
+// Create new room array if a weapon is picked
+const favouriteRoomArray = (rooms) => {
+  if (mystery.weapon) {
+    const onlyFavouriteRooms = fillArray(mystery.weapon, 8);
+    const newRoomArray = [...rooms];
+
+    onlyFavouriteRooms.forEach((item, index) => {
+      newRoomArray.push(onlyFavouriteRooms[index].resident);
+
+      index++;
+    });
+    return newRoomArray;
+  } else {
+    return rooms;
+  }
+};
+
+// Randomly select a favourite weapon from an array
+const shuffleFavouriteWeapon = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+// Remove active from class if it already exists, add if not
+const toggleActive = (front, back) => {
```

Comment:
Cool that you use the toggle-function for this. It makes it feel like you actually draw a new (randomized) card! 😊👌